wordpress-mobile/WordPress-Android
# Fix timezone issue on traffic chart

Fixes #20334

This fixes showing incorrect date range on the bar chart on TRAFFIC tab.

#20362 did not resolve the issue in cases where the site timezone is set to 0 and the device timezone is set to +14.
This 
- Reverts #20362: f06774fa9c836112a74ea7e43cf204c20f365267
- Disables normalization of the timezone while fetching visits. Since we are displaying dates from the device locale, there's no need to normalize the timezone with the site zone. In other words, if the date is Mar 1 on my device but Mar 2 on the site timezone, we should fetch Mar 1. fb4ac4072e000e1b97fcb9b429949747f03c85e7

> [!WARNING]
> Before Merging
> - [x] Merge https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2965
> - [x] Update FluxC version to FluxC's trunk

|before|after|
|-|-|
|<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/526a23c6-96f2-4c01-b395-b4003480423a" width=300 >|<img src="https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/70d70cd6-871e-400e-b1c1-327541d0547d" width=300>|

-----

## To Test:



1. Log in to JP.
2. Change the device timezone to an edge, -14, or -12.
3. Open the Stats Traffic tab.
4. Tap "By day" and change it to "By week".
5. Verify that the date range of the date picker is the same as the one shown on the chart.
6. Change the device timezone to another edge, such as +12. Alternatively, you can change the date of the device.
7. Restart the app.
8. Repeat 3-5.

-----

## Regression Notes

1. Potential unintended areas of impact

    - None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

3. What automated tests I added (or what prevented me from doing so)

    - The issue was caused by a logic from 3rd party lib (FluxC). It's not possible to add automated tests in WPAndroid.

-----

## PR Submission Checklist:


-----

## Testing Checklist:
